### PR TITLE
Add documentation for ToggleAdvanced widget

### PR DESCRIPTION
**Pull request by serhiyzhovnir**
## Purpose of this pull request

This pull request (PR) adds information about the ToggleAdvanced widget.
The PR is an adjusted copy of https://github.com/magento/devdocs/pull/4713 PR.
I've created it on purpose to not to increase repo size too much.

## Affected DevDocs pages

These pages do not exist yet, but they will be created for the ToggleAdvanced widget documentation and the URLs should be like the following:

- https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_toggle.html
- https://devdocs.magento.com/guides/v2.2/javascript-dev-guide/widgets/widget_toggle.html
- https://devdocs.magento.com/guides/v2.1/javascript-dev-guide/widgets/widget_toggle.html

## Links to Magento source code

- https://github.com/magento/magento2/blob/2.1/lib/web/mage/toggle.js - Toggle Widget source code
- https://github.com/magento/magento2/blob/2.2/lib/web/mage/toggle.js - Toggle Widget source code
- https://github.com/magento/magento2/blob/2.3/lib/web/mage/toggle.js - Toggle Widget source code
- https://github.com/magento/magento2/blob/2.1/app/code/Magento/CatalogInventory/view/frontend/templates/stockqty/composite.phtml#L16 - Example of usage
- https://github.com/magento/magento2/blob/2.2/app/code/Magento/CatalogInventory/view/frontend/templates/stockqty/composite.phtml#L16 - Example of usage
- https://github.com/magento/magento2/blob/2.3/app/code/Magento/CatalogInventory/view/frontend/templates/stockqty/composite.phtml#L16 - Example of usage

whatsnew
Partner Atwix added a [new topic](https://devdocs.magento.com/guides/v2.3/javascript-dev-guide/widgets/widget_toggle.html) describing the `ToggleAdvanced` widget.

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
+---
+group: javascript-developer-guide
+subgroup: 3_Widgets
```
Comment (jeff-matthews) [inline guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md]: The `subgroup` property is obsolete and can be deleted.

File: guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+title: ToggleAdvanced widget
+contributor_name: Atwix
+contributor_link: https://www.atwix.com/
+---
+
+## Overview
+
+The ToggleAdvanced {% glossarytooltip f0dcf847-ce21-4b88-8b45-83e1cbf08100 %}widget{% endglossarytooltip %} is used to toggle HTML class for selectors by click on a toggle element.
```
Comment (jeff-matthews) [inline guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md]:
```suggestion
Use the ToggleAdvanced [widget](https://glossary.magedevteam.com/widget) to toggle the HTML class for selectors by clicking on a toggle element.
```

Comment (jeff-matthews) [reply guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md]: We don't use the `glossarytooltip` liquid tag anymore.

File: guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+
+Also, it allows toggling a clicked element class and a clicked element text if an element is toggled.
```
Comment (jeff-matthews) [inline guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md]:
```suggestion
The widget also allows toggling a clicked element class and clicked element text if an element is toggled.
```

File: guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+
+The ToggleAdvanced widget source is [lib/web/mage/toggle.js].
+
+## Initialize the ToggleAdvanced widget
+
+For information about how to initialize a widget in a JS component or `.phtml` template, see the [Initialize JavaScript] topic.
+
+Generally the ToggleAdvanced widget is instantiated like following:
```
Comment (jeff-matthews) [inline guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md]:
```suggestion
To instantiate the ToggleAdvanced widget:
```

File: guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+- [baseToggleClass](#basetoggleclass)
+- [selectorsToggleClass](#selectorstoggleclass)
+- [toggleContainers](#togglecontainers)
+- [newLabel](#newlabel)
+- [curLabel](#curlabel)
+- [currentLabelElement](#currentlabelelement)
+
+### `baseToggleClass`
+
+The class used to be toggled on clicked element.
```
Comment (jeff-matthews) [inline guides/v2.1/javascript-dev-guide/widgets/widget_toggle.md]:
```suggestion
The class used to toggle on a clicked element.
```

**Comment by dobooth** [comment]
@serhiyzhovnir Thanks for the contribution. Due to the EOL for v2.1, we aren't accepting any changes to that version. We can merge this PR after you update your branch and remove the v2.1 file from this PR.

**Comment by serhiyzhovnir** [comment]
Hi @jeff-matthews @dobooth 
We have added the required changes.
Also, we have removed the ToggleAdvanced widget documentation for v2.1.
Thank you!

**Comment by jeff-matthews** [comment]
Thanks @serhiyzhovnir!

**Comment by jeff-matthews** [comment]
running tests

**Comment by jeff-matthews** [comment]
running tests

**Pull request merged**